src/expressions/adaptJavaScriptValueToXPathValue.ts

```diff
@@ -69,16 +71,37 @@ function adaptItemToXPathValue(value: any, domFacade: DomFacade): Value | null {
 					})
 			);
 	}
+	// This code will be reached if the passed value is not actually a ValidValue. This can happen
+	// as JavaScript does not check the types from TypeScript
 	throw new Error(`Value ${value} of type "${typeof value}" is not adaptable to an XPath value.`);
 }
 
+function checkNumericType(value: ValidValue, type: ValueType): value is number {
```

Comment:
I'll take a second to update TypeScript so that we can use the `asserts value is number` syntax this afternoon. I'd prefer the `asserts` syntax if we can use it